### PR TITLE
Cw module

### PR DESCRIPTION
**Pull request by mridulk97**

### CODE REVIEW

File: analysis/generate_with_transformer.py [hunk 1]
```diff
@@ -123,4 +195,5 @@ def generate_images(index, species_true_indx,
     config = OmegaConf.merge(configs, cli)
     print(config)
     
-    main(config)
\ No newline at end of file
+    main(config)
```
Comment (elhamod) [inline analysis/generate_with_transformer.py]: I would rather you separate this into two files. It is generally not nice to keep commenting code to make something work. Doing that in config files is fine, but not code.

File: analysis/model_performance.py [hunk 2]
```diff
@@ -27,6 +27,9 @@ def main(configs_yaml):
     
     # Load model
     config = load_config(yaml_path, display=False)
+    # model = load_CWVQGAN(config, ckpt_path=ckpt_path, data=dataset.data, cuda=(DEVICE is not None))
```
Comment (elhamod) [inline analysis/model_performance.py]: Same here. Either add a flag or make a new file, depending how much code is shared.

File: analysis/test_resnet.py [hunk 3]
```diff
@@ -78,24 +104,51 @@ def main(configs_yaml):
...
+    level0_mapping = [['Alosa chrysochloris', 'Carassius auratus', 'Cyprinus carpio', 'Notropis atherinoides', 'Notropis blennius', 'Notropis boops', 'Notropis buccatus', 'Notropis buchanani', 'Notropis dorsalis', 'Notropis hudsonius', 'Notropis leuciodus', 'Notropis nubilus', 'Notropis percobromus', 'Notropis stramineus', 'Notropis telescopus', 'Notropis texanus', 'Notropis volucellus', 'Notropis wickliffi', 'Noturus exilis', 'Noturus flavus', 'Noturus gyrinus', 'Noturus miurus', 'Noturus nocturnus', 'Phenacobius mirabilis'],
+    ['Esox americanus', 'Gambusia affinis', 'Lepomis auritus', 'Lepomis cyanellus', 'Lepomis gibbosus', 'Lepomis gulosus', 'Lepomis humilis', 'Lepomis macrochirus', 'Lepomis megalotis', 'Lepomis microlophus', 'Morone chrysops', 'Morone mississippiensis'],
+    ['Lepisosteus osseus', 'Lepisosteus platostomus']]
+
+    level_dict = {0: level0_mapping, 1: level1_mapping, 2: level2_mapping}
     preds = []
     classes = []
     for img, target in tqdm(dataloader):
         if phylogeny_path is not None:
             target = phylomapper.get_mapped_truth(target)
```
Comment (elhamod) [inline analysis/test_resnet.py]: Same here

File: configs/archived/custom_vqgan-256img-phylo-vqvae.yaml [hunk 4]
```diff
@@ -47,7 +47,8 @@ model:
           disc_factor: 0.0
           perceptual_weight: 0.0
```
Comment (elhamod) [inline configs/archived/custom_vqgan-256img-phylo-vqvae.yaml]: nit remove empty space changes to simplify review

**Review by elhamod — CHANGES_REQUESTED** [review]

File: analysis/generate_with_transformer.py (hunk 1 above)
Comment (mridulk97) [reply analysis/generate_with_transformer.py]: created a new file generate_with_transformer_cw.py and the original one is unchanged now

File: analysis/test_resnet.py (hunk 3 above)
Comment (mridulk97) [reply analysis/test_resnet.py]: created a new file test_resnet_cw.py and the original one is unchanged now

File: analysis/model_performance.py (hunk 2 above)
Comment (mridulk97) [reply analysis/model_performance.py]: added a config to evaluate between model loading of phly vs CW

File: configs/archived/custom_vqgan-256img-phylo-vqvae.yaml (hunk 4 above)
Comment (mridulk97) [reply configs/archived/custom_vqgan-256img-phylo-vqvae.yaml]: removed !

**Review by elhamod — APPROVED** [review]

**Pull request merged**